### PR TITLE
feat(frontend): group recent matches and player history by date

### PR DESCRIPTION
**Pull request by maantrifork**
## Summary

- Group matches in the recent-matches list on the league page by date. When a match is not from today, a small uppercase header ("Yesterday" or "Wednesday, Mar 5") appears above the first match of each new date bucket.
- Apply the same grouping to the match history on the player page.
- Add `isToday` and `matchDateGroupLabel` helpers in `$lib/utils` for shared use.

image

image

## Why

When the recent-matches list spans more than one day, comparing per-row timestamps is slow. Grouping by date gives users an at-a-glance boundary between today's matches and older buckets.

## Implementation

- Group label is pre-computed in a `$derived` that walks `data.recentMatches` (and `matches` on the player page) and emits `{ match, label | null }` pairs. The previous attempt used `$state` writes inside the each-block template, which Svelte 5 forbids (`state_unsafe_mutation`); the derived approach keeps render pure.
- The header appears only when the match is not today and the previous match was either today or from a different group, so labels never duplicate within a contiguous bucket.

## Test plan

- [ ] Open a league whose most recent matches span today + yesterday + earlier. Confirm today's matches have no header, yesterday's block has a "Yesterday" header, and earlier blocks have a weekday/date header.
- [ ] Open the same player page and confirm the same grouping applies under "Matches".
- [ ] `npm run check` passes locally.

## New Features
* Matches in recent-matches and match history lists are now grouped by date with uppercase headers ("Yesterday" or formatted dates like "Wednesday, Mar 5") displayed above each date group for improved organization.

### CODE REVIEW

**Comment by bjarkehs** [comment]
@coderabbitai review

**Review by bjarkehs — CHANGES_REQUESTED** [review]
I really love the idea. I just want to make sure the logic is sound around the grouping.

**Review by bjarkehs — COMMENTED** [review]
## `/code-review max` — date grouping for match lists

Reviewed in an isolated worktree against `origin/main`, with the real frontend toolchain run. **The logic is sound**: I verified the descending-sort assumption holds end-to-end (backend `OrderByDescending(PlayedAt).ThenByDescending(RecordedAt)` → client `.map`/`.filter` preserve order), keying via `group.match.id` is equivalent to the old `match.id`, the `MatchCard` prop contract is unchanged, and the `match → group.match` rename is complete (flag refs, ring wrapper, `openFlagDialog`). `playedAt` is non-nullable in the TS model, so the helper null-guards are defensive only.

One **blocker**: prettier / `npm run lint` fails in CI (details inline) — `npm run format` fixes it. Everything else inline is optional polish; the duplication + day-key items touch the same code and are worth folding in together. Nice feature 👍

File: frontend/src/routes/(authed)/[orgSlug]/[leagueSlug]/+page.svelte [hunk 1]
```diff
@@ -30,6 +31,20 @@
...
+    (data.recentMatches ?? []).map((match, i, arr) => {
+      const today = isToday(match.playedAt);
+      const prev = arr[i - 1];
+      const newGroup =
+        !today &&
+        (prev == null ||
+          isToday(prev.playedAt) ||
+          matchDateGroupLabel(prev.playedAt) !==
+            matchDateGroupLabel(match.playedAt));
+      return { match, label: newGroup ? matchDateGroupLabel(match.playedAt) : null };
```
Comment (bjarkehs) [inline frontend/src/routes/(authed)/[orgSlug]/[leagueSlug]/+page.svelte]: 🔴 **CI blocker — `npm run lint` fails on this PR.** `test-ui.yml` runs `prettier --check . && eslint .` on every frontend PR, and prettier reports formatting issues in both touched `.svelte` files (eslint never even runs, since prettier exits first). Affected spots:

- this line (44) and `player/[id]/+page.svelte:69` — `return { match, label: … };` exceeds the print width and must wrap onto multiple lines
- line 133 here and `player/[id]/+page.svelte:342` — Tailwind class order (`prettier-plugin-tailwindcss` reorders them)
- `player/[id]/+page.svelte:23` — the `$lib/utils` import is too long to stay on one line

Fix: `cd frontend && npm run format`. (`npm run check` / type-check passes — only formatting fails.)

File: frontend/src/routes/(authed)/[orgSlug]/[leagueSlug]/+page.svelte [hunk 2]
```diff
@@ -30,6 +31,20 @@
   let reportModalOpen = $state(false);
   let ratingHistory = $state<LeagueRatingHistoryEntry[] | undefined>(undefined);
 
+  const recentMatchGroups = $derived(
```
Comment (bjarkehs) [inline frontend/src/routes/(authed)/[orgSlug]/[leagueSlug]/+page.svelte:34]: **Duplicated grouping logic.** This `$derived` block is identical to `matchGroups` in `player/[id]/+page.svelte:59` (only the source array differs), and the uppercase header `<div>` markup is byte-identical in both files. Since you've already extracted `isToday`/`matchDateGroupLabel`, the natural next step is a shared `groupMatchesByDate(matches)` in `$lib/utils` returning `{ match, label }[]`.

Upside: the day-key fix (see the `utils.ts` note) lands in one place, and you can compute each match's day key once — today `matchDateGroupLabel(match.playedAt)` is recomputed up to 3× per row, and on the player page this runs over up to 1000 matches and re-runs on every filter toggle.

File: frontend/src/routes/(authed)/[orgSlug]/[leagueSlug]/player/[id]/+page.svelte [hunk 3]
```diff
@@ -56,6 +56,20 @@
     })
   );
 
+  const matchGroups = $derived(
```
Comment (bjarkehs) [inline frontend/src/routes/(authed)/[orgSlug]/[leagueSlug]/player/[id]/+page.svelte]: Same block as `recentMatchGroups` on the league page (`[leagueSlug]/+page.svelte:34`) — see the note there. Worth extracting into a shared `groupMatchesByDate` helper rather than keeping two copies in sync by hand.

File: frontend/src/lib/utils.ts [hunk 4]
```diff
@@ -17,6 +17,31 @@ export function formatDateTime(d: string | null | undefined): string {
...
+  const date = new Date(d);
+  const now = new Date();
+  return (
+    date.getFullYear() === now.getFullYear() &&
+    date.getMonth() === now.getMonth() &&
+    date.getDate() === now.getDate()
+  );
+}
+
+export function matchDateGroupLabel(d: string | null | undefined): string {
```
Comment (bjarkehs) [inline frontend/src/lib/utils.ts]: **Group boundaries are computed from the display label, which omits the year.** This formats as `weekday, month day` (e.g. `Wednesday, Mar 5`) with no year, and both pages detect a new bucket via `matchDateGroupLabel(prev) !== matchDateGroupLabel(cur)`. Two matches on the same weekday+month+day in different years produce an identical label, so the second is merged into the first's bucket with no header.

Likelihood is low (needs adjacent matches a calendar-repeat cycle apart — only really reachable on the player page's `limit:1000` multi-year history), but the fix is cheap and decouples bucketing from presentation: compare a normalized day key (`new Date(d).toDateString()` or `YYYY-MM-DD`) for boundaries, and keep this function for rendering only. (Same root cause as @coderabbitai's inline note — flagging it at the source.)

File: frontend/src/lib/utils.ts [hunk 5]
```diff
@@ -17,6 +17,31 @@ export function formatDateTime(d: string | null | undefined): string {
   return new Date(d).toLocaleString();
 }
 
+export function isToday(d: string | null | undefined): boolean {
```
Comment (bjarkehs) [inline frontend/src/lib/utils.ts]: Two small notes on these new helpers:

1. **SSR/client timezone skew.** `new Date()` here (and in `matchDateGroupLabel`) runs both during SSR (server TZ, usually UTC) and again on the client (user-local TZ) inside the `$derived`. Near midnight a match can be classified `today` server-side (header suppressed) but `Yesterday` client-side, so a header flashes in on hydration. Low severity, but the PR newly makes a structural present/absent decision based on it.
2. **No unit tests.** These are pure functions and the repo already unit-tests utils under `src/lib/util/*.test.ts` (`isPresent`, `season`, `url`); CI runs `npm run test`. A small test covering today / yesterday / older / null (and the year-collision above) would lock in the edge cases.

**Review by bjarkehs — APPROVED** [review]
LGTM ✅

Reviewed at max effort: `npm run check` is clean, all 15 new unit tests pass, and the `formatDate` consolidation is behavior-preserving across every call site.

One item worth tracking as a follow-up (non-blocking): the date-bucket headers are computed during SSR in the server's timezone (UTC) but re-derived client-side in the viewer's zone, so users far from UTC can briefly see a header flash on load. Easy to address later by computing the labels after mount. Nice work!
